services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,17 +45,55 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+
+  if (!thermalMgrQueueHandle) {
+    return ERR_CODE_INVALID_STATE;
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) != pdTRUE) {
+    return ERR_CODE_QUEUE_FULL;
+  }
+
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  // Send an event to handle the over-temperature and exit the interrupt ASAP
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
   while (1) {
-    
+    thermal_mgr_event_t event;
+
+    if (xQueueReceive(thermalMgrQueueHandle, &event, 0) == pdTRUE) {
+      // An event was received
+      error_code_t errCode;
+      float temp = 0.0f;
+
+      LOG_IF_ERROR_CODE(readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp));
+
+      if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD && errCode) {
```

Comment:
Instead, it is better to add an if(errCode != success) continue; statement instead after each read